browser/ntp_sponsored_images/view_counter_service_factory.cc

```diff
@@ -48,24 +48,29 @@ ViewCounterServiceFactory::~ViewCounterServiceFactory() {}
 
 KeyedService* ViewCounterServiceFactory::BuildServiceInstanceFor(
     content::BrowserContext* browser_context) const {
-  Profile* profile = Profile::FromBrowserContext(browser_context);
+  // Only NTP in normal profile uses sponsored services.
+  if (browser_context->IsOffTheRecord())
+    return nullptr;
 
-  bool is_supported_locale = false;
-  auto* ads_service = brave_ads::AdsServiceFactory::GetForProfile(profile);
-  if (!ads_service) {
-    LOG(ERROR) << "Ads service was disabled at build time!";
-  } else {
-    is_supported_locale = ads_service->IsSupportedLocale();
-  }
+  if (auto* service = g_brave_browser_process->ntp_sponsored_images_service()) {
+    Profile* profile = Profile::FromBrowserContext(browser_context);
+    bool is_supported_locale = false;
+    auto* ads_service = brave_ads::AdsServiceFactory::GetForProfile(profile);
+    if (!ads_service) {
+      LOG(ERROR) << "Ads service was disabled at build time!";
```

Comment:
I know this isn't something you changed, but this also doesn't seem like it should be an error log. Why log anything at all here?